app/models/search_log.rb

```diff
@@ -41,7 +44,9 @@ def self.log(term:, search_type:, ip_address:, user_agent: nil, user_id: nil)
     ip_address = nil if user_id
     key = redis_key(user_id: user_id, ip_address: ip_address)
 
-    user_agent = nil if user_agent && user_agent.length > 2000
+    if user_agent && user_agent.length > MAXIMUM_USER_AGENT_LENGTH
+      user_agent = user_agent[0...MAXIMUM_USER_AGENT_LENGTH]
```

Comment:
```suggestion
      user_agent = user_agent.truncate(MAXIMUM_USER_AGENT_LENGTH)
```

minor nit https://apidock.com/rails/String/truncate